src/OpenTK.GLControl/GLControl.cs

```diff
@@ -191,32 +212,37 @@ protected override void OnHandleCreated(EventArgs e)
                 implementation.WindowInfo.Dispose();
             }
 
-            if (design_mode)
-            {
-                implementation = new DummyGLControl();
-            }
-            else
+            try
             {
-                implementation = new GLControlFactory().CreateGLControl(format, this);
-            }
+                if (design_mode)
+                    implementation = new DummyGLControl();
+                else
+                    implementation = new GLControlFactory().CreateGLControl(format, this);
 
-            context = implementation.CreateContext(major, minor, flags);
-            MakeCurrent();
+                context = implementation.CreateContext(major, minor, flags);
+                MakeCurrent();
 
-            if (!design_mode)
+                if (!design_mode)
```

Comment:
no short ifs

Reply:
Actually this was already in the file.
Resharper is installed on my PC and it actually goes wild on this file.
I will change this.